File: 07/7.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool isPrime(long long n);

int main()
{
    long long count = 0 , i = 1;
    while (count != 10001)
    {
        if(isPrime(i))
            count ++;
        i += 2;
    }
    cout << i-2 << "\n";
}

bool isPrime(long long n)
{
    if (n < 4)
        return true;
    if (n % 2 == 0)
        return false;
    for (long long i = 3 ; i <= (int)pow(n , 0.5) ; i+=2)
    {
        if (n % i == 0)
            return false;
    }
    return true;
}

File: 07/7.java
class main
{
    public static void main (String[] args)
    {
        int i = 1 , count = 0;
        while (count != 10001)
        {
            if (isPrime(i))
                count++;
            i += 2;
        }
        System.out.println("" + (i-2));
    }
    public static boolean isPrime(int n)
    {
        if (n < 4)
            return true;
        if (n % 2 == 0)
            return false;
        for (int i = 3 ; i <= (int)Math.pow(n , 0.5) ; i += 2)
        {
            if (n%i == 0)
                return false;
        }
        return true;
    }
}
